syntax/procfile.vim

```diff
@@ -14,17 +14,21 @@ syn region procfileLine    start='^'      end='$' oneline contains=procfileComme
 syn region procfileComment start='#'      end='$' oneline contained
 syn region procfileBundle  start='bundle' end='$' oneline contained contains=procfileEnvSetting,procfileComment,procfileVariable
 
-syn match procfileName    /^\w\+:/ contained contains=procfileInvalidName,procfileValidName 
-syn region procfileInvalidName start='^' end=':'                          oneline contained
-syn region procfileValidName   start='^' end='\(web\|_worker\|_handler\|_scheduler\|_listener\):' oneline contained
+syn match procfileName    /^[[:alnum:]_-]\+:/ contained contains=procfileInvalidName,procfileValidName
+syn region procfileInvalidName start='^' end=':'                    oneline contained
+syn region procfileValidName   start='^' end='\v\w+[[:alnum:]_-]?:' oneline contained
 
 syn region procfileEnv     start='env' end='$' oneline contained contains=procfileBundle,procfileEnvSetting,procfileEnvProg transparent
 syn keyword procfileEnvProg env contained
 syn match procfileEnvSetting    /\s\S\+=\S\+/ contained contains=procfileEnvAssignment
 syn match procfileEnvAssignment /=\S\+/       contained contains=procfileEnvOperator
 syn match procfileEnvOperator   /=/           contained
 
-syn match procfileVariable /\$\w\+/ contained
+syn match procfileVariable /\v\$\w+/ contained
+" Match variable w/brackets {} around it
+syn match procfileVariable /\v\$\{\w+\}/ contained
+" Match variable w/brackets {} around it and defaulting null/undefined/etc.
+syn match procfileVariable /\v\$\{\w+:[-=?+]\w+\}/ contained
```

Comment:
The leading `\v` is Vim's "very magic" mode, and lets you omit the `\` escape for things that don't usually need escaping in a RegEx. Like the `+`, for example.

Happy to go back to the "normal" way if you prefer.